pkg/oauthserver/userregistry/identitymapper/lookup_test.go

```diff
@@ -25,226 +17,76 @@ func TestLookup(t *testing.T) {
 		},
-		"existing identity, missing user reference": {
-			ProviderName:     "idp",
-			ProviderUserName: "bob",
-
-			ExistingIdentity: makeIdentity("bobIdentityUID", "idp", "bob", "bobUserUID", "bob"),
-			ExistingUser:     nil,
 
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
-				{Name: "GetUser", Object: "bob"},
-			},
-			ExpectedError: true,
-		},
-		"existing identity, invalid user UID reference": {
-			ProviderName:     "idp",
-			ProviderUserName: "bob",
-
-			ExistingIdentity: makeIdentity("bobIdentityUID", "idp", "bob", "bobUserUIDInvalid", "bob"),
-			ExistingUser:     makeUser("bobUserUID", "bob", "idp:bob"),
-
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
-				{Name: "GetUser", Object: "bob"},
-			},
-			ExpectedError: true,
-		},
-		"existing identity, user reference without identity backreference": {
-			ProviderName:     "idp",
-			ProviderUserName: "bob",
-
-			ExistingIdentity: makeIdentity("bobIdentityUID", "idp", "bob", "bobUserUID", "bob"),
-			ExistingUser:     makeUser("bobUserUID", "bob" /*, "idp:bob"*/),
-
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
-				{Name: "GetUser", Object: "bob"},
-			},
-			ExpectedError: true,
-		},
 		"existing identity, user reference": {
 			ProviderName:     "idp",
 			ProviderUserName: "bob",
 
-			ExistingIdentity: makeIdentity("bobIdentityUID", "idp", "bob", "bobUserUID", "bob"),
-			ExistingUser:     makeUser("bobUserUID", "bob", "idp:bob"),
+			// have a user and a mapping
+			ExistingResources: []runtime.Object{
+				makeUserIdentityMapping("bobIdentityUID", "idp", "bob", "bobUserUID", "bob"),
+				makeUser("bobUserUID", "bob", "idp:bob"),
+			},
 
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
-				{Name: "GetUser", Object: "bob"},
-				{Name: "GetUser", Object: "bob"}, // extra request is for group lookup
+			ValidateActions: func(t *testing.T, actions []clienttesting.Action) {
+				if len(actions) != 2 {
+					t.Fatal(spew.Sdump(actions))
+				}
+				if !actions[0].Matches("get", "useridentitymappings") {
+					t.Error(spew.Sdump(actions))
+				}
+				if !actions[1].Matches("get", "users") || actions[1].(clienttesting.GetAction).GetName() != "bob" {
+					t.Error(spew.Sdump(actions))
+				}
 			},
 			ExpectedUserName: "bob",
 		},
 	}
 
 	for k, tc := range testcases {
-		actions := []test.Action{}
-		identityRegistry := &test.IdentityRegistry{
-			GetIdentities: map[string]*userapi.Identity{},
-			Actions:       &actions,
-		}
-		userRegistry := &test.UserRegistry{
-			GetUsers: map[string]*userapi.User{},
-			Actions:  &actions,
-		}
-		if tc.ExistingIdentity != nil {
-			identityRegistry.GetIdentities[tc.ExistingIdentity.Name] = tc.ExistingIdentity
-		}
-		if tc.ExistingUser != nil {
-			userRegistry.GetUsers[tc.ExistingUser.Name] = tc.ExistingUser
-		}
-
-		mappingStorage := mappingregistry.NewREST(userRegistry, identityRegistry)
```

Comment:
this mocking structure was effectively re-running tests from here: https://github.com/openshift/origin/blob/master/pkg/user/registry/useridentitymapping/rest_test.go .  The code in this package relies on the external client.